packages/projection/README.md

```diff
@@ -1,20 +1,100 @@
 # Cardano JS SDK | projection
 
-Chain Sync event projection utilities.
+This package is a library of generic projection types and utilities.
 
-## Summary
+If you're interested in projecting into PostgreSQL database, see [projection-typeorm](../projection-typeorm) package.
 
-Projection is based on [RxJS](https://rxjs.dev/), where source observable of Chain Sync events is processed with various [operators](./src/operators/).
+If you're interested in projector application as run by Lace, see [setup](../cardano-services/src/Projection) and [README](../cardano-services/README.md) in `cardano-services` package.
 
-There are no restrictions what an operator can do - you can utilize the full power of RxJS which makes it very flexible.
+## Prerequisites
 
-All operators implemented in this package are extending the source event object with extra properties, e.g.
+In order to understand how `cardano-js-sdk` projections work, you'll first need basic understanding of:
+- [Observables](https://github.com/tc39/proposal-observable) (we're using [RxJS](https://rxjs.dev/) implementation)
+- Chain Sync mini protocol (well explained in [Ogmios docs](https://ogmios.dev/mini-protocols/local-chain-sync/))
 
-```ts
-Bootstrap.fromCardanoNode({ buffer, cardanoNode, logger }).pipe(
-  Mappers.withStakeKeys(),
-  tap(({ stakeKeys }) => console.log('stakeKeys', stakeKeys)),
-  Mappers.withStakePools(),
-  tap(({ stakePools }) => console.log('stakePools', stakePools))
-);
+## Types
+
+### ProjectionEvent\<ExtraProps>
+
+All projections start by obtaining a source/producer<sup>1</sup>, which is an `Observable<ProjectionEvent<{}>>`. These events are very similar to Chain Sync events, but there are some important differences:
+
+1. Block format is compatible with types from `@cardano-sdk/core` package.
+2. Events include some additional properties: `{eraSumaries, genesisParameters, epochNo, crossEpochBoundary}`.
+3. `RollBackward` events include block data (instead of just specifying the rollback point), and are emitted once for **each** rolled back block.
+
+#### ExtraProps (Generic Parameter)
+
+Source observable can be piped through a series of RxJS operators, which may add some properties to the event. In a nutshell, `ProjectionEvent<T> = ProjectionEvent<{}> & T`. For more detail see [Mappers](#mappers).
+
+### StabilityWindowBuffer
+
+Since `ProjectionEvent{eventType=RollBackward}` includes block data, we must store some number (at least `k`, which is the "security parameter") of latest blocks so that if a fork/rollback happens, we can call `StabilityWindowBuffer.getBlock(hash)` in order to build a valid `ProjectionEvent`.
```

Comment:
I would rephrase it to emphasize that StabilityWindowBuffer is needed for including block data in ProjectionEvent. As it is now, I understand that ProjectionEvent includes block data, which causes storing some number of blocks.
> "In order to include block data in `ProjectionEvent{eventType=RollBackward}`, we must..."
or
> "Previous blocks data must be stored in order to support including block data in `ProjectionEvent{eventType=RollBackward}`. Therefore, we must store...."